src/main/java/seedu/duke/InternshipList.java

```diff
@@ -44,6 +44,15 @@ public Internship getInternship(int index) {
         }
     }
 
+    // Method to update the status of applications, might not be needed
+    public void updateStatus(int index, String status) {
+        if (index >= 0 && index < internships.size()) {
+            internships.get(index).setStatus(status);
+        } else {
+            System.out.println("Invalid index");
+        }
```

Comment:
Looks good overall but wouldn't it be better if instead of setStatus(status) on line 50, we call the new updateStatus(status) function which you have implemented ? In this way, we can check for the invalid input cases also when setting the status. I will merge the changes now but I shall change this line in my next pull request if you give me the permission to do so, thanks !

Reply:
Sure, I understand what you mean, please go ahead.